### PR TITLE
ci: split make and check steps

### PR DESCRIPTION
**Pull request by johelegp**
Addresses #4529 partially and https://github.com/cplusplus/draft/issues/4529#issuecomment-786786718 completely.

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -6,14 +6,20 @@ on:
...
 
+defaults:
+  run:
+    working-directory: source
+
 jobs:
   run-checks:
     name: Run checks
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
Comment (jensmaurer) [inline .github/workflows/check.yml:16]: Why are we fixating this to Ubuntu 20.04 ?  "ubuntu-latest" seems to be less long-term maintenance effort, unless Ubuntu decides to change the package structure.

**Review by jensmaurer — CHANGES_REQUESTED** [review]

File: .github/workflows/check.yml (hunk 1 above)
Comment (johelegp) [reply .github/workflows/check.yml:16]: It's to address this: https://github.com/cplusplus/draft/issues/4529#issuecomment-786624331. I point out what's wrong in the second part of its following comment.

Comment (johelegp) [reply .github/workflows/check.yml:16]: See the first note at https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on. During the migration period, using `runs-on: ubuntu-latest` could use an Ubuntu 18.04 or 20.04 runner. @jwakely's fork workflow runs (https://github.com/jwakely/draft/actions) were all on Ubuntu 20.04 (for example, see https://github.com/jwakely/draft/runs/1986966401?check_suite_focus=true#step:1:4).

Comment (jensmaurer) [reply .github/workflows/check.yml:16]: Ok, thanks.

**Pull request merged**